### PR TITLE
Add xcb functions

### LINKED ISSUE #101: XCB functions
Do you plan to add bindings to https://cairographics.org/manual/cairo-XCB-Surfaces.html, at least in cairo-sys? I need them to implement 29.2.5 at https://www.khronos.org/registry/vulkan/specs/1.0-wsi_extensions/xhtml/vkspec.html#VkXcbSurfaceCreateInfoKHR.

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes #101.

Do you mind giving it a try @Yamakaky please?

### CODE REVIEW

File: cairo-sys-rs/src/lib.rs [hunk 1]
```diff
@@ -55,6 +55,27 @@ pub struct cairo_antialias_t(c_void);
 pub struct cairo_line_join_t(c_void);
 #[repr(C)]
 pub struct cairo_line_cap_t(c_void);
+
+#[cfg(feature = "xlib")]
```
Comment (Yamakaky) [inline cairo-sys-rs/src/lib.rs]: shouldn't it be xcb?

**Comment by Yamakaky** [comment]
Did you forget to add `xcb_window_t`? Also, do you know how I can get a `xcb_visualtype_t` with only a `xcb_connection_t` and a `xcb_window_t`?
The corresponding Xlib code is here (please comment if I did something wrong): https://github.com/Yamakaky/cendre/blob/master/src/surface/xlib.rs#L28-L33

File: src/context.rs [hunk 2]
```diff
@@ -88,25 +90,44 @@ impl Context {
         self.status().ensure_valid();
     }
 
+    /// Creates a new Context with all graphics state parameters set to default values
```
Comment (EPashkin) [inline src/context.rs]: @GuillaumeGomez, Cairo docs not under GPL?

Comment (GuillaumeGomez) [inline src/context.rs]: Oh damn, didn't pay attention that docs were regenerated!

**Comment by GuillaumeGomez** [comment]
> Did you forget to add xcb_window_t? Also, do you know how I can get a xcb_visualtype_t with only a xcb_connection_t and a xcb_window_t?
The corresponding Xlib code is here (please comment if I did something wrong): https://github.com/Yamakaky/cendre/blob/master/src/surface/xlib.rs#L28-L33

I don't know either, that's why I made the inner types public. You can create them yourself from the xlib I suppose.

File: cairo-sys-rs/src/lib.rs (hunk 1 above)
Comment (GuillaumeGomez) [inline cairo-sys-rs/src/lib.rs]: Yes indeed, thanks!

**Comment by GuillaumeGomez** [comment]
So, removed the unwanted docs and added missing files. Sorry about that!

**Comment by Yamakaky** [comment]
SGTM, otherwise.

**Comment by Yamakaky** [comment]
When do you want to merge this?
Also there are still some functions missing, like Quartz. https://www.cairographics.org/manual/cairo-surfaces.html

**Comment by GuillaumeGomez** [comment]
When you confirm it works for you I guess? We don't have much interest in it so you're the one to tell us in here. ;)

**Comment by GuillaumeGomez** [comment]
Oh and could you open an issue for the other one as well please?

**Comment by Yamakaky** [comment]
Since I couldn't find how to get an `xcb_visualtype_t` and XLib works, I didn't check. As it's only function signatures, it should be fine though.

**Comment by GuillaumeGomez** [comment]
Fine then, let's merge!

**Pull request merged**